sky/spot/controller.py

```diff
@@ -294,11 +263,31 @@ def _handle_signal(job_id):
         f'User sent {user_signal.value} signal.')
 
 
+def _cleanup(job_id: int, task_yaml: str):
+    # NOTE: The code to get cluster name is same as what we did in the spot
+    # controller, we should keep it in sync with SpotController.__init__()
+    task_name = pathlib.Path(task_yaml).stem
+    task = sky.Task.from_yaml(task_yaml)
```

Comment:
Just a reminder, in our master branch, we have changed the way we get the task name. Please refer to https://github.com/skypilot-org/skypilot/blob/3654d72208ec0cdc66ca34a98862988de7744810/sky/spot/controller.py#L34-L35